litestar/serialization/msgspec_hooks.py

```diff
@@ -194,71 +205,61 @@ def encode_msgpack(obj: Any, enc_hook: Callable[[Any], Any] | None = default_ser
         SerializationException: If error encoding ``obj``.
     """
     try:
-        if enc_hook is None or enc_hook is default_serializer:
-            return _msgspec_msgpack_encoder.encode(obj)
-        return msgspec.msgpack.encode(obj, enc_hook=enc_hook)
+        if serializer is None or serializer is default_serializer:
+            return _msgspec_msgpack_encoder.encode(value)
+        return msgspec.msgpack.encode(value, enc_hook=serializer)
     except (TypeError, msgspec.EncodeError) as msgspec_error:
         raise SerializationException(str(msgspec_error)) from msgspec_error
 
 
 @overload
-def decode_msgpack(raw: bytes) -> Any:
+def decode_msgpack(value: bytes) -> Any:
     ...
 
 
 @overload
-def decode_msgpack(raw: bytes, type_: type[T]) -> T:
+def decode_msgpack(value: bytes, type_decoders: TypeDecodersSequence | None) -> Any:
     ...
 
 
-def decode_msgpack(raw: bytes, type_: Any = Empty) -> Any:
-    """Decode a MessagePack string/bytes into an object.
-
-    Args:
-        raw: Value to decode
-        type_: An optional type to decode the data into
+@overload
+def decode_msgpack(value: bytes, target_type: type[T]) -> T:
+    ...
 
-    Returns:
-        An object
 
-    Raises:
-        SerializationException: If error decoding ``raw``.
-    """
-    try:
-        if type_ is Empty:
-            return _msgspec_msgpack_decoder.decode(raw)
-        return msgspec.msgpack.decode(raw, dec_hook=dec_hook, type=type_)
-    except msgspec.DecodeError as msgspec_error:
-        raise SerializationException(str(msgspec_error)) from msgspec_error
+@overload
+def decode_msgpack(value: bytes, target_type: type[T], type_decoders: TypeDecodersSequence | None) -> T:
+    ...
 
 
-def decode_media_type(raw: bytes, media_type: MediaType | str, type_: Any) -> Any:
-    """Decode a raw value into an object.
+def decode_msgpack(value: bytes, target_type: type[T] | EmptyType = Empty, type_decoders: TypeDecodersSequence | None = None) -> Any:  # type: ignore[misc]
```

Comment:
note in general here - i am not sure the use of overloads serves us. In the end we get a lot of typing issues still because of this implementation pattern. We can just drop the overloads and generic and return Any, which is pretty standard for dump/load logic.